build.gradle.kts

```diff
@@ -1,44 +1,80 @@
 plugins {
-    kotlin("jvm") version "1.4.31"
-    kotlin("kapt") version "1.4.31"
-    kotlin("plugin.allopen") version "1.4.31"
-    id("io.micronaut.application") version "1.4.0"
+    id("org.jetbrains.kotlin.jvm") version "1.6.10"
+    id("org.jetbrains.kotlin.kapt") version "1.6.10"
+    id("org.jetbrains.kotlin.plugin.allopen") version "1.6.10"
+    id("com.github.johnrengelman.shadow") version "7.1.1"
+    id("io.micronaut.application") version "3.1.1"
 }
 
+version = "0.1"
+group = "hello"
+
+val kotlinVersion = project.properties.get("kotlinVersion")
```

Comment:
I prefer not using this and instead using the `implementation(kotlin("reflect"))` syntax below.